packages/actor-http-fetch/lib/ActorHttpFetch.ts

```diff
@@ -76,12 +76,46 @@ export class ActorHttpFetch extends ActorHttp {
 
     try {
       requestInit = await this.fetchInitPreprocessor.handle(requestInit);
+      let retryCount: number = action.context?.get(KeysHttp.httpRetryCount) ?? 0;
 
-      // Perform request
       const customFetch: ((input: RequestInfo, init?: RequestInit) => Promise<Response>) | undefined = action
         .context?.get(KeysHttp.fetch);
 
-      const response = await (customFetch || fetch)(action.input, requestInit);
+      // Actual fetch respecting timeout and retry count.
+      const getResponse = async(): Promise<Response> => {
+        // When retry count is greater than 0, record the errors for failed requests.
+        const errors: unknown[] = [];
+        while (retryCount-- >= 0) {
+          try {
+            return await (customFetch || fetch)(action.input, requestInit);
+          } catch (error: unknown) {
+            // If the fetch was aborted by timeout, we won't retry.
+            if (requestInit.signal?.aborted) {
+              throw error;
+            }
+            errors.push(error);
+            // Wait for specified delay.
+            if (retryCount >= 0) {
```

Comment:
Won't this always be true?

Reply:
Not on the last allowed try in which case the loop is subsequently exited and and the encountered error is thrown.